setup.py

```diff
@@ -5,22 +5,66 @@
 
 # -*- coding: utf-8 -*-
 
-import os
-
 import setuptools
+import subprocess
+
+from distutils.cmd import Command
+from pathlib import Path
+from shutil import rmtree
+from os import makedirs
 
-this = os.path.dirname(__file__)
+self_dir = Path(__file__).parent
 
 packages = setuptools.find_packages()
 assert packages
 
-README = os.path.join(os.getcwd(), "README.md")
+README = self_dir.joinpath("README.md").resolve()
 with open(README, encoding="utf-8") as f:
     long_description = f.read()
     start_pos = long_description.find("## Contributing")
     if start_pos >= 0:
         long_description = long_description[:start_pos]
 
+
+class OpgenCommand(Command):
+    description = "Generate the static ONNX opset classes"
+    user_options = []
+
+    def initialize_options(self): pass
+    def finalize_options(self): pass
+
+    def run(self):
+        base_path = self_dir.joinpath("onnxscript", "onnx_opset")
+        try:
+            rmtree(base_path)
+        except FileNotFoundError:
+            pass # if base_path doesn't exist, that's great
+
+        # need to generate a blank onnx_opset module since
+        # onnxscript/__init__.py will import it (and we deleted it above);
+        # it will be overridden with correct code as part of the generation
+        # below.
+        makedirs(base_path)
+        with open(base_path.joinpath("__init__.py"), "w"):
+            pass
+
+        from onnxscript.onnx_opset_builder import OpsetsBuilder
+        builder = OpsetsBuilder()
+        paths = builder.write(base_path)
+        subprocess.run(["black", "--quiet", *paths])
```

Comment:
If you use black, it may be better not to add the opset files to the repo. It may create unrelated changes to PR if a different version of black is used.